collectors/proc.plugin/proc_net_wireless.c

```diff
@@ -0,0 +1,501 @@
+        last_node->next = wireless;
+    }
+    else
+        netwireless_root = wireless;
+
+    return wireless;
+}
+
+static bool configure_device(int do_status, int do_quality, int do_discarded_packets, int do_missed,
+							 struct netwireless *wireless_dev) {
+	wireless_dev->do_status = do_status;
+	wireless_dev->do_quality = do_quality;
+	wireless_dev->do_discarded_packets = do_discarded_packets;
+	wireless_dev->do_missed_beacon = do_missed;
+	wireless_dev->configured = true;
+
+	char buffer[RRD_ID_LENGTH_MAX + 1];
+
+	snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s", wireless_dev->name);
+	wireless_dev->chart_id_net_status = strdupz(buffer);
+
+	snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s", wireless_dev->name);
+	wireless_dev->chart_id_net_link = strdupz(buffer);
+
+	snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s", wireless_dev->name);
+	wireless_dev->chart_id_net_level = strdupz(buffer);
+
+	snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s", wireless_dev->name);
+	wireless_dev->chart_id_net_noise = strdupz(buffer);
+
+	snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s", wireless_dev->name);
+	wireless_dev->chart_id_net_discarded_packets = strdupz(buffer);
+
+	snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s", wireless_dev->name);
+	wireless_dev->chart_id_net_missed_beacon = strdupz(buffer);
+
+	return true;
+}
+
+int do_proc_net_wireless(int update_every, usec_t dt) {
+	(void)dt;
+	static procfile *ff = NULL;
+	static int do_status = -1, do_quality = -1, do_discarded_packets = -1, do_missed = -1;
+	static unsigned long long int dt_to_refresh_speed = 0;
+	static char *proc_net_wireless_filename = NULL;
+	static int enable_new_interfaces = -1;
+
+	if(unlikely(enable_new_interfaces == -1)) {
+		char filename[FILENAME_MAX + 1];
+		snprintfz(filename, FILENAME_MAX, "%s", "/proc/net/wireless");
+
+		proc_net_wireless_filename = config_get(CONFIG_SECTION_PLUGIN_PROC_NETWIRELESS,
+												"filename to monitor", filename);
+        enable_new_interfaces = config_get_boolean_ondemand(CONFIG_SECTION_PLUGIN_PROC_NETWIRELESS,
+															"enable new interfaces detected at runtime",
+															CONFIG_BOOLEAN_AUTO);
+
+        do_status = config_get_boolean_ondemand(CONFIG_SECTION_PLUGIN_PROC_NETWIRELESS,
+													  "status for all interfaces", CONFIG_BOOLEAN_AUTO);
+
+        do_quality = config_get_boolean_ondemand(CONFIG_SECTION_PLUGIN_PROC_NETWIRELESS,
+													  "quality for all interfaces", CONFIG_BOOLEAN_AUTO);
+
+        do_discarded_packets = config_get_boolean_ondemand(CONFIG_SECTION_PLUGIN_PROC_NETWIRELESS,
+															  "discarded packets for all interfaces",
+															  CONFIG_BOOLEAN_AUTO);
+
+        do_missed = config_get_boolean_ondemand(CONFIG_SECTION_PLUGIN_PROC_NETWIRELESS,
+													  "missed for all interfaces", CONFIG_BOOLEAN_AUTO);
+
+        dt_to_refresh_speed = config_get_number(CONFIG_SECTION_PLUGIN_PROC_NETWIRELESS,
+												"refresh interface speed every seconds", 10) * USEC_PER_SEC;
```

Comment:
It would be good to add the module description and configuration to the [documentation](https://github.com/netdata/netdata/blob/master/collectors/proc.plugin/README.md).

Reply:
It was added yesterday to the PR.